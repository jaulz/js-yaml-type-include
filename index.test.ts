import fs from 'fs'
import yaml from 'js-yaml'
import createType from './index'

const fixture = fs.readFileSync('./fixtures/include.txt', 'utf8')

test('includes content of existing file', () => {
  const type = createType()
  const schema = new yaml.Schema({
    include: [yaml.DEFAULT_SAFE_SCHEMA],
    explicit: [type],
  })
  const content = yaml.load(
    `
includedFile: !!include "fixtures/include.txt"
  `,
    { schema }
  )

  expect(content).not.toBeNull()
  expect(content.includedFile).not.toBeNull()
  expect(content.includedFile).toEqual(fixture)
  expect(content).toMatchSnapshot('load')
  expect(yaml.dump(content)).toMatchSnapshot('dump')
})

test('includes content of existing file with custom path', () => {
  const type = createType({ relativeTo: './fixtures' })
  const schema = new yaml.Schema({
    include: [yaml.DEFAULT_SAFE_SCHEMA],
    explicit: [type],
  })
  const content = yaml.load(
    `
includedFile: !!include "include.txt"
  `,
    { schema }
  )

  expect(content).not.toBeNull()
  expect(content.includedFile).not.toBeNull()
  expect(content.includedFile).toEqual(fixture)
  expect(content).toMatchSnapshot('load')
  expect(yaml.dump(content)).toMatchSnapshot('dump')
})

test('throw error for non-existing file', () => {
  const type = createType()
  const schema = new yaml.Schema({
    include: [yaml.DEFAULT_SAFE_SCHEMA],
    explicit: [type],
  })

  expect(() => {
    yaml.load(
      `
includedFile: !!include "fixtures/nonExistant.txt"
  `,
      { schema }
    )
  }).toThrowErrorMatchingSnapshot()
})
