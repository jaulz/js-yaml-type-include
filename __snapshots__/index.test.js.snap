// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`includes content of existing file with custom path: dump 1`] = `
"includedFile: Custom Include
"
`;

exports[`includes content of existing file with custom path: load 1`] = `
Object {
  "includedFile": "Custom Include",
}
`;

exports[`includes content of existing file: dump 1`] = `
"includedFile: Custom Include
"
`;

exports[`includes content of existing file: load 1`] = `
Object {
  "includedFile": "Custom Include",
}
`;

exports[`throw error for non-existing file 1`] = `
"cannot resolve a node with !<tag:yaml.org,2002:include> explicit tag at line 2, column 51:
     ... clude \\"fixtures/nonExistant.txt\\"
                                         ^"
`;
